bundle/edu.gemini.oodb.too.url/src/main/java/edu/gemini/spdb/rapidtoo/impl/TooUpdateServiceImpl.java

```diff
@@ -220,78 +218,77 @@ private ISPObsComponent _findComponent(ISPObservation obs, SPComponentType type)
-    private void _setTarget(ISPObservation obs, TooUpdate update)  {
-        TooTarget tooTarget = update.getBasePosition();
+    private void _setTarget(final ISPObservation obs, final TooUpdate update)  {
+        final TooTarget tooTarget = update.getBasePosition();
         if (tooTarget == null) {
             LOG.warning("Too update missing target information.");
             return;
         }
 
         // Set the title of the observation.
-        SPObservation obsDobj = (SPObservation) obs.getDataObject();
+        final SPObservation obsDobj = (SPObservation) obs.getDataObject();
         obsDobj.setTitle(tooTarget.getName());
 
         // Get the target component and update the base position.
-        ISPObsComponent targetComp = _findComponent(obs, TargetObsComp.SP_TYPE);
-        TargetObsComp targetObsComp = (TargetObsComp) targetComp.getDataObject();
-        TargetEnvironment targetEnv = targetObsComp.getTargetEnvironment();
-        SPTarget base = targetEnv.getBase();
+        final ISPObsComponent targetComp = _findComponent(obs, TargetObsComp.SP_TYPE);
+        if (targetComp == null) {
+            LOG.warning("Too update has no target component.");
+            return;
+        }
+
+        final TargetObsComp targetObsComp = (TargetObsComp) targetComp.getDataObject();
+        final TargetEnvironment targetEnv = targetObsComp.getTargetEnvironment();
 
+        final SPTarget base = targetEnv.getBase();
         base.setName(tooTarget.getName());
         base.setRaDecDegrees(tooTarget.getRa(), tooTarget.getDec());
         base.setMagnitudes(tooTarget.getMagnitudes());
 
         // Set the guide star, if present.
-        TooGuideTarget gs = update.getGuideStar();
+        final TooGuideTarget gs = update.getGuideStar();
         if (gs != null) {
-            TooGuideTarget.GuideProbe tooProbe = gs.getGuideProbe();
+            final TooGuideTarget.GuideProbe tooProbe = gs.getGuideProbe();
             if (tooProbe == null) {
                 LOG.warning("Guide star probe not specified.");
             } else {
-                Double ra  = gs.getRa();
-                Double dec = gs.getDec();
-
-                if ((ra == null) || (dec == null)) {
-                    LOG.warning("Guide star ra or dec missing.");
-                } else {
-                    // damn
-                    GuideProbe probe = null;
-                    switch (tooProbe) {
-                        case AOWFS:
-                            probe = AltairAowfsGuider.instance;
-                            break;
-                        case OIWFS:
-                            probe = getOiwfs(obs);
-                            break;
-                        case PWFS1:
-                            probe = PwfsGuideProbe.pwfs1;
-                            break;
-                        case PWFS2:
-                            probe = PwfsGuideProbe.pwfs2;
-                            break;
-                    }
+                // damn
```

Comment:
Haha! I think that was already there regarding the cases, and just got moved or realigned. I'm really not sure, but I think we can probably take it out :-).